### PR TITLE
Ability to upload duplicate file

### PR DESCRIPTION
**Pull request by ghost**
Possible to upload a file if the same filename is available in the server. For example, if a file is available as banner.jpg then new file will be uploaded as banner1.jpg

### CODE REVIEW

**Comment by Bakual** [comment]
i'm not a fan of changing the name of the uploaded file without asking the user. The user may not even notice that the name got changed and references a wrong file afterwards.

Imho it's better to just a proper error and let the user decide if he wants to rename the file or replace the existing one (deleting it first).

File: administrator/components/com_media/controllers/file.json.php [hunk 1]
```diff
@@ -133,19 +133,58 @@ public function upload()
...
-				$response = array(
-					'status'   => '0',
-					'message'  => JText::_('COM_MEDIA_ERROR_FILE_EXISTS'),
-					'error'    => JText::_('COM_MEDIA_ERROR_FILE_EXISTS'),
-					'location' => str_replace(JPATH_ROOT, '',  $filepath)
-				);
+				// Create duplicate destination
+				$i = 0;
+                do {
+                    $base_name  = JFile::stripExt($object_file->filepath) . ($i ? "$i" : "");
```
Comment (dgt41) [inline administrator/components/com_media/controllers/file.json.php]: I think dash `-` or underscore `_` is better than a dot `.`...

Comment (JoomShaper) [inline administrator/components/com_media/controllers/file.json.php]: What about organise folder by date? Like 2015/oct/26/banner.jpg?

Comment (dgt41) [inline administrator/components/com_media/controllers/file.json.php]: This is way to opinionated to go in the core, unless we offer more options (user, categories etc)
It is way too easy to do that with a plugin: https://github.com/dgt41/imageUploaderHelper

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on ba9aaa7c47733fed97d546fb66044e99dfc10df2<br /><br />Applied patch and still could not upload a duplicate file<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8536">issues.joomla.org/joomla-cms/8536</a>.</sub>

**Comment by brianteeman** [comment]
@joomshaper please can you review your code as currently it doesnt work at all. If no response is received then the PR will be closed in a few weeks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8536">issues.joomla.org/joomla-cms/8536</a>.</sub>

**Comment by brianteeman** [comment]
Closed as stated above<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8536">issues.joomla.org/joomla-cms/8536</a>.</sub>

**Pull request closed without merging**